### PR TITLE
Limit store textfields to max 4096 chars

### PR DESCRIPTION
**Pull request by phlax**
there doesnt seem to be any limit on size of text/comment fields atm, which can cause problems

### CODE REVIEW

File: pootle/apps/pootle_store/models.py [hunk 1]
```diff
@@ -198,7 +201,9 @@ class Unit(models.Model, base.TranslationUnit):
     store = models.ForeignKey("pootle_store.Store", db_index=True,
                               on_delete=models.CASCADE)
     index = models.IntegerField(db_index=True)
-    unitid = models.TextField(editable=False)
+    unitid = models.TextField(
+        editable=False,
+        validators=[MaxLengthValidator(4096)])
```
Comment (unho) [inline pootle/apps/pootle_store/models.py]: Not sure if this is actually necessary. Is it possible to edit it from client side?

Comment (phlax) [inline pootle/apps/pootle_store/models.py]: it doesnt really matter - it needs a limit - otherwise things fail

Comment (phlax) [inline pootle/apps/pootle_store/models.py]: not sure what that limit should be tho

File: pootle/apps/pootle_store/models.py [hunk 2]
```diff
@@ -214,10 +219,22 @@ class Unit(models.Model, base.TranslationUnit):
...
+        blank=True,
+        validators=[MaxLengthValidator(4096)])
+    translator_comment = models.TextField(
+        null=True,
+        blank=True,
+        validators=[MaxLengthValidator(4096)])
+    locations = models.TextField(
+        null=True,
+        editable=False,
+        validators=[MaxLengthValidator(4096)])
```
Comment (unho) [inline pootle/apps/pootle_store/models.py]: This field is not editable from client side. I would say leave it out.

Comment (phlax) [inline pootle/apps/pootle_store/models.py]: as above

**Comment by dwaynebailey** [comment]
To give you some reassurance http://www.gamefaqs.com/boards/922345-section-z/64202650

At 4096 if you are translating in one unit you're definitely doing something wrong.

File: pootle/apps/pootle_store/models.py [hunk 3]
```diff
@@ -214,10 +219,22 @@ class Unit(models.Model, base.TranslationUnit):
...
-    locations = models.TextField(null=True, editable=False)
-    context = models.TextField(null=True, editable=False)
+    developer_comment = models.TextField(
+        null=True,
+        blank=True,
+        validators=[MaxLengthValidator(4096)])
+    translator_comment = models.TextField(
+        null=True,
+        blank=True,
+        validators=[MaxLengthValidator(4096)])
```
Comment (unho) [inline pootle/apps/pootle_store/models.py]: I would further restrict this. 4096 characters for a translator comment is like a whole novel.

**Pull request closed without merging**